From faf94821fdf4c8ded45f0a972fbf21deb0d69629 Mon Sep 17 00:00:00 2001
From: ljia5 <lin.a.jia@intel.com>
Date: Wed, 12 Dec 2012 14:20:28 +0800
Subject: [PATCH 35/38] [PORT FROM r4.1] Change
 bionic/libc/kernel/common/linux/psb_drm.h same as the
 kernel file psb_drm.h

BZ: 74310

The file bionic/libc/kernel/common/linux/psb_drm.h is different from he kernel
 file psb_drm.h, so change this file.

Change-Id: I6ca416c5727acede3bdc79806f3c9083abdbbb9e
Signed-off-by: ljia5 <lin.a.jia@intel.com>
Reviewed-on: http://android.intel.com:8080/80945
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Qiu, Junhai <junhai.qiu@intel.com>
Reviewed-by: Zhu, Tianyang <tianyang.zhu@intel.com>
Tested-by: Tong, BoX <box.tong@intel.com>
Reviewed-by: Tong, BoX <box.tong@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 libc/kernel/common/linux/psb_drm.h |   17 ++++++++++++++++-
 1 files changed, 16 insertions(+), 1 deletions(-)

diff --git a/libc/kernel/common/linux/psb_drm.h b/libc/kernel/common/linux/psb_drm.h
index 095c1fc..0b9284e 100644
--- a/libc/kernel/common/linux/psb_drm.h
+++ b/libc/kernel/common/linux/psb_drm.h
@@ -44,6 +44,17 @@
 #define DRM_PSB_FLAG_MEM_MMU_TILING TTM_PL_FLAG_PRIV3
 typedef int32_t psb_fixed;
 typedef uint32_t psb_ufixed;
+
+static inline int32_t psb_int_to_fixed(int a)
+{
+ return a * (1 << PSB_FIXED_SHIFT);
+}
+
+static inline uint32_t psb_unsigned_to_ufixed(unsigned int a)
+{
+ return a << PSB_FIXED_SHIFT;
+}
+
 typedef enum {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  DRM_CMD_SUCCESS,
@@ -805,13 +816,17 @@ typedef struct drm_psb_msvdx_decode_status {
 #define DRM_PSB_ENABLE_IED_SESSION 0x30
 #define DRM_PSB_DISABLE_IED_SESSION 0x31
 #define DRM_PSB_VSYNC_SET 0x32
+/* HDCP */
+#define DRM_PSB_HDCP_DISPLAY_IED_OFF        0x33
+#define DRM_PSB_HDCP_DISPLAY_IED_ON         0x34
+#define DRM_PSB_QUERY_HDCP_DISPLAY_IED_CAPS 0x35
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
 #define DRM_OEM_RESERVED_START 0x40
 #define DRM_OEM_RESERVED_END 0x4F
 #define DRM_PSB_TTM_START 0x50
 #define DRM_PSB_TTM_END 0x5F
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
-#ifdef PDUMP
+#if defined(PDUMP)
 #define DRM_PSB_CMDBUF (PVR_DRM_DBGDRV_CMD + 1)
 #else
 #define DRM_PSB_CMDBUF (DRM_PSB_TTM_START)
-- 
1.7.9

